go/vt/tabletserver/sysloglogger/sysloglogger.go

```diff
@@ -0,0 +1,63 @@
+// Package sysloglogger implements an optional plugin that logs all queries to syslog.
+package sysloglogger
+
+import (
+	"flag"
+	"fmt"
+	"log/syslog"
+
+	log "github.com/golang/glog"
+	"github.com/youtube/vitess/go/vt/servenv"
+	"github.com/youtube/vitess/go/vt/tabletserver"
+)
+
+// syslogWriter is an interface that wraps syslog.Writer, so it can be mocked in unit tests.
+type syslogWriter interface {
+	Info(string) error
+	Close() error
+}
+
+// writer holds a persistent connection to the syslog daemon (or a mock when under test).
+var writer syslogWriter
+
+// ch holds the tabletserver.StatsLogger channel to which this plugin subscribes (or a mock when under test).
+var ch chan interface{}
+
+// logQueries is the vttablet startup flag that must be set for this plugin to be active.
+var logQueries = flag.Bool("log_queries", false, "Enable query logging to syslog.")
+
+func init() {
+	servenv.OnRun(func() {
+		if *logQueries {
+			var err error
+			writer, err = syslog.New(syslog.LOG_INFO, "vtquerylogger")
+			if err != nil {
+				log.Errorf("Query logger is unable to connect to syslog: %v", err)
+				return
+			}
+			go run()
+		}
+	})
+}
+
+// Run logs queries to syslog, if the "log_queries" flag is set to true when starting vttablet.
+func run() {
+	defer writer.Close()
+
+	// ch will only be non-nil in a unit test context, when a mock has been populated
+	if ch == nil {
+		ch := tabletserver.StatsLogger.Subscribe("gwslog")
+		defer tabletserver.StatsLogger.Unsubscribe(ch)
+	}
+
+	formatParams := map[string][]string{"full": {}}
+	for out := range ch {
```

Comment:
Depending on the performance of the underlying syslog writer, you may want to have multiple go routines all reading from the ch and writer to syslog. But I wouldn't do it until you notice some messages are eaten up...